plugins/woocommerce/includes/class-wc-structured-data.php

```diff
@@ -304,46 +304,55 @@ public function generate_product_data( $product = null ) {
 					$min_sale_price = min( $child_sale_prices );
 				}
 
+				$unit_price_specification = array(
+					'@type'                 => 'UnitPriceSpecification',
+					'price'                 => wc_format_decimal( $min_price, wc_get_price_decimals() ),
+					'priceCurrency'         => $currency,
+					'valueAddedTaxIncluded' => wc_prices_include_tax(),
+					'validThrough'          => $price_valid_until,
+				);
+				if ( $product->is_on_sale() && $min_price !== $min_sale_price ) {
+					// `priceType` should only be specified in prices which are not the current offer.
+					// https://developers.google.com/search/docs/appearance/structured-data/merchant-listing#sale-pricing-example
+					$unit_price_specification['priceType'] = 'https://schema.org/ListPrice';
+				}
 				$markup_offer = array(
 					'@type'              => 'Offer',
 					'priceSpecification' => array(
-						array(
-							'@type'                 => 'UnitPriceSpecification',
-							'priceType'             => 'https://schema.org/ListPrice',
-							'price'                 => wc_format_decimal( $min_price, wc_get_price_decimals() ),
-							'priceCurrency'         => $currency,
-							'valueAddedTaxIncluded' => wc_prices_include_tax(),
-							'validThrough'          => $price_valid_until,
-						),
+						$unit_price_specification,
 					),
 				);
 
-				if ( $product->is_on_sale() ) {
```

Comment:
There was the chance that the a Groupped product could have:

1. Product A with regular price 18.
2. Product B with regular price 20 and sale price 18.

That caused the structured data to show the regular price and the sale price as 18. I added a check to make sure the prices are different.